libs/iavl/mutable_tree_oec.go

```diff
@@ -164,15 +169,20 @@ func (tree *MutableTree) setNewWorkingTree(version int64, persisted bool) ([]byt
 	}
 	treeMap.updatePpnc(version)
 
+	tree.updatePruningVersion()
```

Comment:
It's a little different with previous logical. now every height will prune?  If so the `historyStateNum` will unuseless? I think we should be careful here.

Reply:
The logic of prune is still controlled by ac and historyStateNum